### DISCUSSION TITLE
Automate the Query to AS and dump in csv

### DISCUSSION THREAD

**Discussion by joaoluisbm** [DaxStudio/DaxStudio#522] [Q&A]
Hello!

So, I was playing around with DAX Studio and I was able to connect to my Tabular Model in SSAS, write some query and then save it in a .csv file.

Now, I'm wondering and trying to find a solution on how I can automate this. Do you have any suggestion? How technology DAX Studio uses behind the scenes to do that?

Thanks a lot!
Best Regards,

**Comment by dgosbell** [comment]
DAX Studio is using the Microsoft client library called Microsoft.AnalysisServices.AdomdClient to connect to tabular servers and run queries. You can call this yourself from either a .Net program or PowerShell. 

I have a short example of how to call this library from Powershell and output the results to a csv [here](https://gist.github.com/dgosbell/75a4c2b6c1eae6ba1a8f4fdddd995119)

**Reply by joaoluisbm** [reply]
SUPER MEGA TKS!

That's exactly what I was looking for! I was digging in google but I couldn't found this reference.

Many many thanks Darrell (ps: great fan of you work here!)

Cheers,
João Martins
